### PR TITLE
Turn off warnings reported by some compiler settings

### PR DESCRIPTION
**Pull request by barendgehrels**

### CODE REVIEW

File: include/boost/geometry/strategies/cartesian/intersection.hpp [hunk 1]
```diff
@@ -226,14 +226,14 @@ struct cartesian_segments
...
 
-            typedef typename promote_integral<CoordinateType>::type promoted_type;
-
-            promoted_type const numerator
-                = boost::numeric_cast<promoted_type>(ratio.numerator());
-            promoted_type const denominator
-                = boost::numeric_cast<promoted_type>(ratio.denominator());
-            promoted_type const dx_promoted = boost::numeric_cast<promoted_type>(dx);
-            promoted_type const dy_promoted = boost::numeric_cast<promoted_type>(dy);
+            typedef typename promote_integral<CoordinateType>::type integral_type;
```
Comment (awulkiew) [inline include/boost/geometry/strategies/cartesian/intersection.hpp]: This type may not be integral type. The purpose of this util is to promote only integral types to bigger integral types and leave other types as they were.

See: https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/util/promote_integral.hpp#L312

Comment (barendgehrels) [reply include/boost/geometry/strategies/cartesian/intersection.hpp]: Thanks I'll rename it. The point was that promoted_type shadows the outer type with the same name.
I can call it `bigger_type` for example

Comment (barendgehrels) [reply include/boost/geometry/strategies/cartesian/intersection.hpp]: Done

**Comment by barendgehrels** [comment]
@awulkiew  can we still merge this one to 1.71?

**Comment by awulkiew** [comment]
@barendgehrels This is not a complex change so I think the managers will allow it.

**Comment by awulkiew** [comment]
@barendgehrels master branch is open for bugfixes. Are you going to cherry pick this yourself and ask the managers for the permission to merge or would you like me to do it?

**Comment by barendgehrels** [comment]
Hi @awulkiew , thank you. I wanted to do it this weekend but I was quite busy preparing a new PR w.r.t. rescaling (to be expected next weekend).
If you could do it, that would be appreciated.

**Comment by awulkiew** [comment]
@barendgehrels It's merged. Would you like to add some release notes? I plan to go through pull requests and issues and add some of them.

**Comment by barendgehrels** [comment]
@awulkiew  Thanks for merging! And for the release notes, no I currently I don't have anything to add there.

**Pull request merged**